src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
+const DEFAULT_HEALTH = 100;
+const BITE_DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name, health = DEFAULT_HEALTH) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= BITE_DAMAGE;
+    }
+
+    if (animal.health <= 0) {
+      Animal.alive = Animal.alive.filter((subject) => subject.health > 0);
```

Comment:
[CODE PERFORMANCE] - Instead of checking for health and then filtering the `Animal.alive` array inside the `bite` method, you could improve the performance by filtering the `Animal.alive` array only when necessary, after the health of an animal is decreased. This avoids unnecessary filtering when the animal's health is still above 0. Consider moving the filter operation outside of the `if` block, so it only runs when an animal's health has actually reached 0 or below.